contracts/AstarBase.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./DappsStaking.sol";
+
+/// @author The Astar Network Team
+/// @title Astarbase. Mapping of Stakers ss58 <> H160
+contract AstarBase is Ownable {
+    using Counters for Counters.Counter;
+    Counters.Counter public registeredCnt;
+
+    bool public paused;
+    bool public onChainCheckEnabled;
+    address FREE_ADDRESS = address(0);
```

Comment:
Can you teach me what this is?

Reply:
- `paused`  Paused is design pattern used in contracts if you want to stop usage. In general you need this for Attacks and if you find major bugs.
- `onChainCheckEnabled`  If we ever want to stop checking on-chain data and only use database entries we can avoid checking it on-chain. 
- For now, `onChainCheckEnabled` this is used until we fix the precompile with the issue in the description

Reply:
Sorry, I was referring to `address FREE_ADDRESS = address(0);` GitHub didn't make it intuitive 😓

Reply:
@hoonsubin by default whole ETH storage for all possible mappings is pointing to 0.
`address(0)` is short for 0x0000000...00
By setting it to 0, you mark it as unused storage

Reply:
ah, thanks for clarifying this for me.